test/metabase/api/user_test.clj

```diff
@@ -641,6 +642,28 @@
                    (change-user-via-api! {:first_name "Blue"
                                           :last_name  nil})))))))))
 
+(deftest update-sso-user-test
+  (testing "PUT /api/user/:id"
+    (testing "Test that we do not update a user's first and last names if they are an SSO user."
+      (mt/with-temp User [{user-id :id} {:first_name   "SSO"
+                                         :last_name    "User"
+                                         :email        "sso-user@metabase.com"
+                                         :sso_source   "jwt"
+                                         :is_superuser true}]
+        (letfn [(change-user-via-api! [m]
+                  (mt/user-http-request :crowberto :put 400 (str "user/" user-id) m))]
+          (testing "`:first_name` changes are rejected"
+            (is (= {:errors {:first_name "Editing first name is not allowed for SSO users."}}
+                   (change-user-via-api! {:first_name "NOT-SSO"}))))
+          (testing "`:last_name` changes are rejected"
+            (is (= {:errors {:last_name "Editing last name is not allowed for SSO users."}}
+                   (change-user-via-api! {:last_name "USER"}))))
+          (testing "New names that are the same as existing names succeed because there is no change."
+            (is (partial= {:first_name "SSO"
+                           :last_name  "User"}
+                          (mt/user-http-request :crowberto :put 200 (str "user/" user-id) {:first_name "SSO"
+                                                                                           :last_name  "User"})))))))))
```

Comment:
Adding the response code as parameter for `hange-user-via-api!` would let you reuse it here too, I think.